### ISSUE TITLE
Lazy loading read receipts

### ISSUE THREAD

**Issue by ara4n** [matrix-org/matrix-doc#1319]
Currently we send an `m.room.receipt` event for each and every member of a room in the initial sync - e.g. old Matrix HQ had 16,000 read receipt objects bulking out the initial sync.

Just as we have the option of lazy loading members over at https://github.com/matrix-org/matrix-doc/issues/1227, we should have a separate option for lazy loading read receipts to save sync bandwidth.

The semantics would be to only send RRs for the events which are returned in the current piece of timeline.

**Comment by non-Jedi** [comment]
I'm less familiar with the ephemeral events side of things, so pardon if this is a silly question. Given that semantically, ephemeral events aren't supposed to be things that are part of the overarching DAG, would it be possible to simply spec that servers may discard ephemeral events at will? Or that servers will only keep the last X% of number of joined room members m.room.receipt events?

**Comment by ara4n** [comment]
@non-Jedi sorry for missing the question at the time. sure, you could just ditch the RRs at random, but wouldn't it be nice to ditch the ones which are off the page rather than on it?  It's also unclear whether EDU's ephemerality semantics are right for RRs, which end up being pretty important for a robust room.

**Comment by ara4n** [comment]
We'd also need to think how to include RRs in /messages or /context results.  It almost feels like we'd want to include them as decoration on an event rather than in a separate block.

**Comment by ara4n** [comment]
will be fixed (at last) by sync v3 (cc @kegsay), i think?

**Comment by kegsay** [comment]
Yes, though it won't affect `/messages` or `/context`, only `/sync`.
